trojsten/contests/views.py

```diff
@@ -84,8 +85,8 @@ def view_pdf(request, round_id, solution=False):
     if solution and not round.solutions_are_visible_for_user(request.user):
         raise Http404
     path = round.get_pdf_path(solution)
-    if os.path.exists(path):
-        response = sendfile(request, path)
+    if settings.TASK_STATEMENTS_STORAGE.exists(path):
+        response = sendfile(request, settings.TASK_STATEMENTS_STORAGE.path(path))
```

Comment:
sendfile only works with local filesystem paths.., this will break if `TASK_STATEMENTS_STORAGE` is set to anything other than `FileSystemStorage`.., we need to read the file manually and return it as a response

Reply:
Is it ok to just check the path does not raise exception?
According to the api, it should return _the local filesystem path_.

We can redirect to the (obfuscated) file url in case the path is not implemented.
This will make the statements accessible from the web by anybody, but shouldn't be a problem unless we leak the url.

Reply:
fallbacking to url if path is not available is a good idea..
external storage systems (like s3, gcs) support signed links == generated URLs by `django-storages`  contain token, that is only valid for a configurable period of time, that way even leaking should not pose a problem..

Reply:
Done.